openvisualizer/moteProbe/moteProbe.py

```diff
@@ -21,18 +21,51 @@
 import time
 import sys
 
+import paho.mqtt.client as mqtt
+import json
+import Queue
+
 from   pydispatch import dispatcher
 import OpenHdlc
 import openvisualizer.openvisualizer_utils as u
 from   openvisualizer.moteConnector import OpenParser
 from   openvisualizer.moteConnector.SerialTester import SerialTester
 
-#============================ functions =======================================
+#============================ defines =========================================
+
+BROKER_ADDRESS              = "argus.paris.inria.fr"
+NOTIFY_SERIALBYTES_TOPIC    = 'opentestbed/deviceType/mote/deviceId/+/notif/frommoteserialbytes'
+TOTAL_NUMBER_MOTES_TESTBED  = 76
+DISCOVER_MOTES_TIMEOUT      = 30
 
 BAUDRATE_LOCAL_BOARD  = 115200
 BAUDRATE_IOTLAB       = 500000
 
-def findSerialPorts(isIotMotes=False):
+#============================ variables =======================================
+
+
+testbedmotes = []
+discovermotes_queue = Queue.Queue()
+
+#============================ functions =======================================
+
+def on_connect(client, userdata, flags, rc):
+
+    print "connect to :", NOTIFY_SERIALBYTES_TOPIC
+    client.subscribe(NOTIFY_SERIALBYTES_TOPIC)
+
+
+def on_message(client, userdata, message):
+
+    # get the motes list from payload
+    
+    testbedmotes.append(message.topic.split('/')[4])
+    
+    if len(testbedmotes) == TOTAL_NUMBER_MOTES_TESTBED:
+        discovermotes_queue.put('all motes responsed')
+
+
+def findSerialPorts(isIotMotes=False, isTestbedMotes=False):
```

Comment:
`isOpentestbedMotes`

Reply:
rework the `findSerialPorts` helper function by splitting it in 3 functions (serial, IoT-lab, OpenTestbed).